pkg/fleet/internal/cdn/cdn.go

```diff
@@ -50,87 +46,110 @@ type orderConfig struct {
+		0,
+		// Not using the roots; send the highest seen version of roots so don't received them all on every request
+		c.currentRootsVersion,
+		// Not using a client cache; fetch all the applicable target files every time.
+		[]*pbgo.TargetFileMeta{},
+	)
 	if err != nil {
 		return nil, err
 	}
-	options := []remoteconfig.Option{
-		remoteconfig.WithAPIKey(c.env.APIKey),
-		remoteconfig.WithConfigRootOverride(c.env.Site, ""),
-		remoteconfig.WithDirectorRootOverride(c.env.Site, ""),
-		remoteconfig.WithDatabaseFileName("remote-config-cdn-tmp"),
+
+	// We always expect configurations.
+	if agentConfigUpdate == nil {
+		return nil, fmt.Errorf("no configurations found for AGENT_CONFIG")
 	}
-	service, err := remoteconfig.NewService(
-		config,
-		"Datadog Installer",
-		fmt.Sprintf("https://config.%s", c.env.Site),
-		hostname,
-		getHostTags(ctx, config),
-		&rctelemetryreporterimpl.DdRcTelemetryReporter{}, // No telemetry for this client
-		version.AgentVersion,
-		options...,
-	)
-	if err != nil {
-		return nil, err
+
+	// Update CDN root versions
+	for _, root := range agentConfigUpdate.TUFRoots {
+		var signedRoot data.Signed
+		err = json.Unmarshal(root, &signedRoot)
+		if err != nil {
+			continue
+		}
+		var r data.Root
+		err = json.Unmarshal(signedRoot.Signed, &r)
+		if err != nil {
+			continue
+		}
+		if uint64(r.Version) > c.currentRootsVersion {
+			c.currentRootsVersion = uint64(r.Version)
+		}
 	}
-	service.Start()
-	defer func() { _ = service.Stop() }()
-	// Force a cache bypass
-	cfgs, err := service.ClientGetConfigs(ctx, &pbgo.ClientGetConfigsRequest{
-		Client: &pbgo.Client{
-			Id:            uuid.New().String(),
-			Products:      []string{"AGENT_CONFIG"},
-			IsUpdater:     true,
-			ClientUpdater: &pbgo.ClientUpdater{},
-			State: &pbgo.ClientState{
-				RootVersion:    1,
-				TargetsVersion: 1,
-			},
-		},
-	})
-	if err != nil {
-		return nil, err
+
+	var signedTargets data.Signed
+	err = json.Unmarshal(agentConfigUpdate.TUFTargets, &signedTargets)
+	if err == nil {
+		var targets data.Targets
+		err = json.Unmarshal(signedTargets.Signed, &targets)
+		if err == nil && uint64(targets.Version) > c.currentTargetsVersion {
+			c.currentTargetsVersion = uint64(targets.Version)
+		}
 	}
 
 	// Unmarshal RC results
 	configLayers := map[string]*layer{}
 	var configOrder *orderConfig
 	var layersErr error
-	for _, file := range cfgs.TargetFiles {
-		matched := datadogConfigIDRegexp.FindStringSubmatch(file.GetPath())
+	paths := agentConfigUpdate.ClientConfigs
+	targetFiles := agentConfigUpdate.TargetFiles
+	for _, path := range paths {
+		matched := datadogConfigIDRegexp.FindStringSubmatch(path)
 		if len(matched) != 2 {
-			layersErr = multierr.Append(layersErr, fmt.Errorf("invalid config path: %s", file.GetPath()))
+			layersErr = multierr.Append(layersErr, fmt.Errorf("invalid config path: %s", path))
 			continue
 		}
 		configName := matched[1]
 
+		file := targetFiles[path]
```

Comment:
Can we add a case to register and error and skip the rest if the key isn't in the map?